From 41c09a670a16ec8284607a5d47fa3f39396d94e6 Mon Sep 17 00:00:00 2001
From: Hui Tang <tanghui20@huawei.com>
Date: Sat, 21 May 2022 18:02:14 +0800
Subject: [PATCH 2150/2592] [Huawei] sched: fix kfree for invalid address

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186820

-------------------------------

We should kfree p->prefer_cpus before kfree p.

Signed-off-by: Hui Tang <tanghui20@huawei.com>
---
 kernel/fork.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/fork.c b/kernel/fork.c
index 88056facda35..f932fd8b4bd5 100644
--- a/kernel/fork.c
+++ b/kernel/fork.c
@@ -464,10 +464,10 @@ void free_task(struct task_struct *tsk)
 	arch_release_task_struct(tsk);
 	if (tsk->flags & PF_KTHREAD)
 		free_kthread_struct(tsk);
-	free_task_struct(tsk);
 #ifdef CONFIG_QOS_SCHED_DYNAMIC_AFFINITY
 	sched_prefer_cpus_free(tsk);
 #endif
+	free_task_struct(tsk);
 }
 EXPORT_SYMBOL(free_task);
 
-- 
2.17.1

